Feat/hero

RodionClepa opened the pull request:
<img width="1847" height="724" alt="image" src="https://github.com/user-attachments/assets/4757179f-6120-411c-b4d5-3a86d8af25f6" />

Commit:
feat(hero): add hero section

Commit:
feat(hero): add infinite carousel for stats

Commit:
fix(hero): handle the flicking in carousel

Commit:
feat(hero): add js for carousel to remove code duplication

Commit:
fix(hero): fix animation pause

Commit:
fix(hero): add specific class for carousel block

Commit:
fix(hero): move starting animation in js logic

Commit:
feat(hero): add mixins for media queries

Commit:
refactor(hero): change the logic with layout

Commit:
feat(hero): add padding variable and utility

Commit:
fix(hero): minor fixes

dmamaliga commented on scss/layouts/_hero.scss:
```diff
@@ -0,0 +1,82 @@
+  }
+
+  &__image-container {
+    background-image: url("../../img/images/hero-image.png");
```
Let's use an `<img>` tag. We should use css only for decorative images that have no meaning.

dmamaliga commented on index.html:
```diff
@@ -1,12 +1,190 @@
+            <span class="form-input__error">Filed erorr. Please fix it</span>
+          </div>
+          <div class="partnership">
+            <p class="grey-2">Our Partnership</p>
```
I would make this an `h2`

RodionClepa replied:
`Our Partnership` is much more smaller than others h2
<img width="337" height="423" alt="Screenshot 2025-08-21 at 11 17 34" src="https://github.com/user-attachments/assets/97d63fcf-c4c9-4194-b52e-e9504917e863" />

dmamaliga commented on index.html:
```diff
@@ -1,12 +1,190 @@
+              where it will be easier for you
+            </p>
+          </div>
+          <div class="form-input form-input--location" for="location">
```
Location search input should be inside a `<form>`.

dmamaliga commented on index.html:
```diff
@@ -1,12 +1,190 @@
+              <li class="partnership__item">
+                <img
+                  class="partnership__logo"
+                  alt="Partner Logo"
```
Ideally we should have more meaningful alt text

dmamaliga commented on index.html:
```diff
@@ -1,12 +1,190 @@
+                class="form-input__field input-location__field"
+                type="location"
+                id="location"
+                placeholder="Search for the location you want!"
```
Your location input is missing a `<label>`. While it has a placeholder, placeholders are not a replacement for labels. A screen reader won't announce the purpose of the input without a proper label. The for attribute you have on a `<div>` is invalid; it must be on a `<label>` and point to the id of an input

dmamaliga commented on index.html:
```diff
@@ -1,12 +1,190 @@
+              </svg>
+              <input
+                class="form-input__field input-location__field"
+                type="location"
```
invalid input type, I think we should use `type="text"` or `type="search"`

dmamaliga commented on index.html:
```diff
@@ -1,12 +1,190 @@
+              <button class="btn btn--green input-location__search-button">
+                Search
+                <svg class="btn__icon">
+                  <use xlink:href="img/icons/sprite.svg#Arrow"></use>
```
The `xlink:href` attribute for SVG `<use>` tags is deprecated. Modern browsers support using just `href`.
See: https://developer.mozilla.org/en-US/docs/Web/SVG/Reference/Attribute/xlink:href

dmamaliga commented on index.html:
```diff
@@ -1,12 +1,190 @@
-    </body>
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
```
let's add a meta description to improve SEO

dmamaliga commented on index.html:
```diff
@@ -1,12 +1,190 @@
+                />
+              </li>
+              <li class="partnership__item">
+                <img
```
We can drastically improve performance by using next gen image formats like webp. If you have time and will, you can convert `png's` and `jpeg's` to `webp's` and use them in the project.

Commit:
fix(hero): solve pr comments

dmamaliga commented on scss/layouts/_hero.scss:
```diff
@@ -0,0 +1,88 @@
+
+  &__image-container {
+    position: relative;
+    min-height: 720px;
```
We need a `max-height` to prevent this.

<img width="1329" height="937" alt="Image" src="https://github.com/user-attachments/assets/58fb4789-21b4-4747-bc32-ec450b690a6d" />

Commit:
fix(hero): add max width for hero image

dmamaliga commented:
Drugoe dela
<img width="500" height="159" alt="image" src="https://github.com/user-attachments/assets/f8b58e27-48c9-4959-badc-bdd99adc643a" />

dmamaliga commented on index.html:
```diff
@@ -1,12 +1,207 @@
+            </p>
+          </div>
+          <form action="#">
+            <div class="form-input form-input--location" for="location">
```
please remove `for` attribute from `div`

Commit:
fix(hero): remove `for` from div`

dmamaliga commented on index.html:
```diff
@@ -1,12 +1,207 @@
+      </nav>
+    </header>
+    <main>
+      <section class="hero section-center">
```
You forgot about this gradient

<img width="772" height="597" alt="Image" src="https://github.com/user-attachments/assets/9559141e-4b25-4756-b4d1-fe1668881b60" />

RodionClepa replied:
I added it in header branch

RodionClepa committed:
Feat/recommendation (#4)

* feat(recommendation): add basic html and css for slider

* feat(recommendation): add basic mvc

* refactor(recommendations): change pub and sub to event type relationship

* feat(recommendations): add base parent class views

* feat(recommendation): add hidden slider item to prevent flickering

* feat(recommendation): add draggable slider

* refactor(recommendations): add handle for last slider item

* feat(recommendation): add debounce utility

* feat(recommendation): add next and prev button for slider

* feat(recommendation): add separate pub-sub class

* feat(recommendation): add slider's button update on scroll

* fix(recommendation): fix button on slider and incorrect calculation for last card

* refactor(recommendation): change the logic of identifying current card

* refactor(recommendation): rename isNearEnd to isEnd

* fix(recommendation): remove draggable image by browser

* refactor(recommendation): change the displaying method

* refactor(recommendation): add the padding for slider

* feat(recommendation): add possibility to view all estates

* fix(recommendation): remove stroke from sprite

* feat(recommendation): add separate Model class

* fix(recommendation): fix slider for safari browser

* fix(recommendation): change slide card to button

* feat(recommendation): change padding layout

* fix(recommendation): change the logic of indentifying the end

* feat(recommendation): add utility class

* feat(recommendation): add padding end

* fix(recommendation): minor fixes

* feat(recommendation): convert images to webp

* fix(recommendation): fix isEnd function in slider class

* fix(recommendation): add aria label for buttons

* Feat/ready to sell (#5)

* feat(ready-to-sell): add main html structure for section

* feat(ready-to-sell): add css for information part

* feat(ready-to-sell): add style for images

* feat(ready-to-sell): change description layout

* fix(ready-to-sell): change icons sprite

* fix(ready-to-sell): minor fixes

* feat(ready-to-sell): convert images to webp

* feat(ready-to-sell): add size for play icon

* feat(ready-to-sell): fix pr comments

* feat(ready-to-sell): add blur

* fix(ready-to-sell): change blur image

* fix(ready-to-sell): change blur image 2

* fix(ready-to-sell): fix overflow of blur image

* Feat/reviews (#6)

* feat(reviews): add basic html and css for slideshow

* feat(reviews): add scrolling and basic mvc structure

* feat(reviews): add view render for slide show

* feat(reviews): add scrolling with button dots

* feat(reviews): add slideshow by default to be in middle

* fix(reviews): fix box-shadow problem for slide item

* fix(reviews): remove comments

* feat(reviews): add hide-scrollbar class for container

* fix(reviews): minor fixes

* feat(reviews): convert images

* fix(reviews): fix text cutting in slider

* fix(reviews): fix overflow issue

* fix(reviews): fix lighthouse complaints

* fix(reviews): pr comments

* fix(reviews): fix typo

* Feat/articles (#7)

* feat(articles): add basic html and css

* feat(articles): add mvc for article list

* feat(articles): add preview article

* feat(articles): add limit to more articles button

* feat(articles): add scroll into view for more articles

* fix(articles): change image displaying

* fix(articles): add object fit cover for blog image

* fix(articles): minor fixes

* fix(articles): fix css overriding

* feat(articles): convert images to webp

* feat(articles): add basic html and css

* fix(articles): pr comments

* fix(articles): add lazy loading for user avatar

* fix(articles): remove console logs

* feat(articles): add margin top for section

* Feat/subscribe (#12)

* feat(recommendation): add basic html and css for slider

* feat(recommendation): add basic html and css for slider

* feat(subscribe): add subscribe section

* fix(subscribe): add wrapper for images for responsivness

* feat(subscribe): convert images to webp

* feat(subscribe): change background to image

* fix(subscribe): fix images

* Feat/form (#9)

* feat(recommendation): add basic html and css for slider

* feat(subscribe): add subscribe section

* feat(form): add form

* feat(form): add separate class for containers

* fix(form): minor fixes

* fix(form): convert images and fix aria issues

* fix(form): pr comments

* fix(form): revert changes and add id to label form

* Feat/footer (#10)

* feat(subscribe): add subscribe section

* feat(footer): add footer

* feat(footer): add blur

* feat(footer): add cookie banner

* fix(footer): add accessibility

* fix(footer): pr comments

* fix(footer): change blur image

* feat(header): add header and blur image (#11)

* feat(header): add header and blur image

* feat(header): add hover effect for links

* fix(header): fix aria complaints

* fix(header): fix blur position

* fix(header): change blur image

---------

Co-authored-by: Rodion Clepa <rclepa>

---------

Co-authored-by: Rodion Clepa <rclepa>

---------

Co-authored-by: Rodion Clepa <rclepa>

---------

Co-authored-by: Rodion Clepa <rclepa>

---------

Co-authored-by: Rodion Clepa <rclepa>

---------

Co-authored-by: Rodion Clepa <rclepa>

---------

Co-authored-by: Rodion Clepa <rclepa>

---------

Co-authored-by: Rodion Clepa <rclepa>
